elixir/apps/api/test/api/controllers/actor_group_controller_test.exs

```diff
@@ -0,0 +1,203 @@
+      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
+    end
+
+    test "lists all actor groups", %{conn: conn, account: account, actor: actor} do
+      actor_groups = for _ <- 1..3, do: Fixtures.Actors.create_group(%{account: account})
+
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> get("/v1/actor_groups")
+
+      assert %{
+               "data" => data,
+               "metadata" => %{
+                 "count" => count,
+                 "limit" => limit,
+                 "next_page" => next_page,
+                 "prev_page" => prev_page
+               }
+             } = json_response(conn, 200)
+
+      assert count == 3
+      assert limit == 50
+      assert is_nil(next_page)
+      assert is_nil(prev_page)
+
+      data_ids = Enum.map(data, & &1["id"]) |> MapSet.new()
+      actor_group_ids = Enum.map(actor_groups, & &1.id) |> MapSet.new()
+
+      assert MapSet.equal?(data_ids, actor_group_ids)
+    end
+
+    test "lists actor groups with limit", %{conn: conn, account: account, actor: actor} do
+      actor_groups = for _ <- 1..3, do: Fixtures.Actors.create_group(%{account: account})
+
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> get("/v1/actor_groups", limit: "2")
+
+      assert %{
+               "data" => data,
+               "metadata" => %{
+                 "count" => count,
+                 "limit" => limit,
+                 "next_page" => next_page,
+                 "prev_page" => prev_page
+               }
+             } = json_response(conn, 200)
+
+      assert limit == 2
+      assert count == 3
+      refute is_nil(next_page)
+      assert is_nil(prev_page)
+
+      data_ids = Enum.map(data, & &1["id"]) |> MapSet.new()
+      actor_group_ids = Enum.map(actor_groups, & &1.id) |> MapSet.new()
+
+      assert MapSet.subset?(data_ids, actor_group_ids)
+    end
+  end
+
+  describe "show" do
+    test "returns error when not authorized", %{conn: conn, account: account} do
+      actor_group = Fixtures.Actors.create_group(%{account: account})
+      conn = get(conn, "/v1/actor_groups/#{actor_group.id}")
+      assert json_response(conn, 401) == %{"errors" => %{"detail" => "Unauthorized"}}
+    end
+
+    test "returns a single actor group", %{conn: conn, account: account, actor: actor} do
+      actor_group = Fixtures.Actors.create_group(%{account: account})
+
+      conn =
+        conn
+        |> authorize_conn(actor)
+        |> put_req_header("content-type", "application/json")
+        |> get("/v1/actor_groups/#{actor_group.id}")
+
+      assert json_response(conn, 200) == %{
+               "data" => %{
+                 "id" => actor_group.id,
+                 "name" => actor_group.name
+               }
+             }
+    end
+  end
+
+  describe "create" do
```

Comment:
It would be also nice to add tests for validation errors (at least what happens when `params` are empty).

Reply:
Yeah, good call.  Will add those types of tests in.

Reply:
Ah, wait, don't I already have those tests.  Line 113 in this file for example.

Reply:
While I did have tests for empty attrs under the given attribute key, I went back and added tests with no params/body at all.